lib/rspec/support/comparable_version.rb

```diff
@@ -0,0 +1,35 @@
+module RSpec
+  module Support
+    # @private
+    class ComparableVersion
+      include Comparable
+
+      attr_reader :string
+
+      def initialize(string)
+        @string = string
+      end
+
+      def <=>(other)
+        other = self.class.new(other) unless other.is_a?(self.class)
+
+        return 0 if string == other.string
+
+        longer_segment_count = [self, other].map { |version| version.segments.count }.max
+
+        longer_segment_count.times do |index|
+          self_segment = segments[index] || 0
+          other_segment = other.segments[index] || 0
+          result = self_segment <=> other_segment
+          return result unless result == 0
+        end
+
+        0
+      end
+
+      def segments
+        @segments ||= string.split('.').map { |v| v.to_i }
+      end
+    end
+  end
+end
```

Comment:
This class looks nice and simple, but will it handle version numbers with `.pre`, `.rc1`, `.beta`, etc in the name?  The `segments` method uses `to_i` which will silently convert those to `0`.  Looking at the [JRuby downloads page](http://jruby.org/files/downloads/index.html), I see version numbers like `9.0.0.0.pre1` and `9.0.0.0.rc1`.  Since we pass the `JRUBY_VERSION` to this I think we need to handle those versions properly.